packages/itwinui-react/src/core/Table/actionHandlers/selectHandler.ts

```diff
@@ -190,7 +191,10 @@ const getSelectedData = <T extends Record<string, unknown>>(
 ) => {
   const selectedData: T[] = [];
   const setSelectedData = (row: Row<T>) => {
-    if (selectedRowIds[row.id]) {
+    // Check whether the row selected is a sub-component.
+    // If so, exclude it from the selected data.
+    const isMainRow = row.original[iuiId as any] === undefined;
+    if (selectedRowIds[row.id] && isMainRow) {
```

Comment:
This *is* the source of truth. The reducer calls this code (indirectly) to generate the resulting state.

Reply:
If so, should I not add subcomponent into all the handler functions that have to do with `newState` in `selectHandler`?

Reply:
I agree with the outcome @r100-stack is describing, but not the implementation. I still think `selectHandler.ts` is the best place for this code, because that's where we are calculating the state returned by the reducer.

If you look at the `onSelectHandler` code, it modifies `newState.selectedRowIds`. So we need to make sure this object does not contain any of the duplicated rows.

https://github.com/iTwin/iTwinUI/blob/6ee8ddede0a4eb1a372b590132d4d8f2b4ca5116/packages/itwinui-react/src/core/Table/actionHandlers/selectHandler.ts#L63-L66

Reply:
I have added a condition whether the sub-rows should be looped through if they are sub-components. This includes the possibility that `onSelect` returns duplicate data so I have removed the piece of code I had originally.